lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +490,1307 @@ LocaleMatch lookupMatcher(
+    return vm::ExecutionStatus::EXCEPTION;
+  }
+  auto optionsToUse = optionsTo.getValue();
+  // 3. Let opt be a new Record.
+  Impl::newRecord opt;
+  // 4. Let matcher be ? GetOption(options, "localeMatcher", "string", «
+  //   "lookup", "best fit" », "best fit").
+  auto matcher = getOptionString(
+      optionsToUse,
+      u"localeMatcher",
+      {u"lookup", u"best fit"},
+      llvh::Optional<std::u16string>(u"best fit"),
+      runtime);
+  // 5. Set opt.[[localeMatcher]] to matcher.
+  opt.localeMatcher = matcher->getValue();
+
+  // 6. Let calendar be ? GetOption(options, "calendar", "string",
+  //    undefined, undefined).
+  auto calendar = getOptionString(
+      optionsToUse, u"calendar", {}, llvh::Optional<std::u16string>(), runtime);
+  // 7. If calendar is not undefined, then
+  //    a. If calendar does not match the Unicode Locale Identifier type
+  //       nonterminal, throw a RangeError exception.
+  // 8. Set opt.[[ca]] to calendar.
+  if (calendar == vm::ExecutionStatus::EXCEPTION) {
+    return runtime->raiseRangeError("Incorrect calendar information provided");
+  }
+  if (calendar->hasValue()) {
+    opt.ca = calendar->getValue();
+  }
+  // 9. Let numberingSystem be ? GetOption(options, "numberingSystem",
+  //    "string", undefined, undefined).
+  auto numberingSystem = getOptionString(
+      optionsToUse,
+      u"numberingSystem",
+      {},
+      llvh::Optional<std::u16string>(),
+      runtime);
+  // 10. If numberingSystem is not undefined, then
+  //     a. If numberingSystem does not match the Unicode Locale Identifier
+  //        type nonterminal, throw a RangeError exception.
+  // 11. Set opt.[[nu]] to numberingSystem.
+  if (numberingSystem == vm::ExecutionStatus::EXCEPTION) {
+    return runtime->raiseRangeError(
+        "Incorrect numberingSystem information provided");
+  }
+  if (numberingSystem->hasValue()) {
+    opt.nu = numberingSystem->getValue();
+  }
+  // 12. Let hour12 be ? GetOption(options, "hour12", "boolean",
+  //     undefined, undefined).
+  auto hour12 =
+      getOptionBool(optionsToUse, u"hour12", llvh::Optional<bool>(), runtime);
+  if (hour12 == vm::ExecutionStatus::EXCEPTION) {
+    return vm::ExecutionStatus::EXCEPTION;
+  }
+  // 13. Let hourCycle be ? GetOption(options, "hourCycle", "string", «
+  //     "h11", "h12", "h23", "h24" », undefined).
+  auto hourCycle = getOptionString(
+      optionsToUse,
+      u"hourCycle",
+      {u"h11", u"h12", u"h23", u"h24"},
+      llvh::Optional<std::u16string>(),
+      runtime);
+  // 14. If hour12 is not undefined, then
+  //     a. Let hourCycle be null.
+  if (hour12->hasValue()) {
+    if (hourCycle->hasValue()) {
+      hourCycle->getValue() = u"";
+      // 15. Set opt.[[hc]] to hourCycle.
+      opt.hc = hourCycle->getValue();
+    }
+  }
+
+  // 16 - 23
+  static const std::vector<std::u16string> relevantExtensionKeys = {
+      u"ca", u"nu", u"hc"};
+  auto r = resolveLocale(
```

Comment:
As I understand, according to https://tc39.es/ecma402/#sec-initializedatetimeformat **17.**, `opt` should be used here instead of `options`.